Added a validation class for notes, plus some tests

dglebovskiy opened the pull request.

dglebovskiy committed:
Added a validation class for notes, plus some tests

alekseyC64 commented on notes_app/notes/api.py:
```diff
@@ -77,6 +78,16 @@ def delete_detail(self, object_list, bundle):
+class NoteValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        errors = {}
+        if len(bundle.data[u'title']) == 0:
```
check http://python-future.org/unicode_literals.html, strings w/o `u''` is cleaner

alekseyC64 replied:
I think validation of title and content should be wider. If they can't be empty, you can check them as `if not bundle.data['title']:` - it will check for null and empty string

dglebovskiy committed:
Imported unicode literals in tests.py
Squashed unauthenticated behavior and note deletion tests together
Added tests for posting notes with absent title/content
Made validation more general

dglebovskiy committed:
Tests for Note PATCH

dglebovskiy committed:
Negative test cases for patch

alekseyC64 commented on notes_app/notes/tests.py:
```diff
@@ -211,6 +211,18 @@ def test_patch_detail(self):
+        """Authenticated user shouldn't be able to update notes belonging
+            to other users"""
+        self.assertHttpUnauthorized(self.api_client.patch(
+            self.api_detail_shared, format='json',
```
should it return `HttpUnauthorized` on a shared note?
